File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/05.While-Loop/02-exercise/04.walking.js
function walking(input) {

    const goal = 10000;

    let index = 0;
    let current = input[index++];
    let totalSteps = 0;

    while (current !== "Going home") {
        let currentSteps = Number(current);
        totalSteps += currentSteps;
        if (totalSteps >= goal) {
            break;
        }

        current = input[index++];
    }

    if (current === "Going home") {
        let stepsToHome = Number(input[index]);
        totalSteps += stepsToHome;
    }

    let difference = Math.abs(totalSteps - goal);
    if (totalSteps >= goal) {
        console.log("Goal reached! Good job!");
        console.log(`${difference} steps over the goal!`);
    } else {
        console.log(`${difference} more steps to reach goal.`);
    }
}
walking(["125",
"250",
"4000",
"30",
"2678",
"4682"]);

File: Python_Basics/Homeworks-And-Labs/05.while_loop/02.exercise/04.walking.py
goal = 10000

current = input()
total_steps = 0

while current != "Going home":
    current_steps = int(current)
    total_steps += current_steps
    if total_steps >= goal:
        break
        
    current = input()
    
if current == "Going home":
    steps_to_home = int(input())
    total_steps += steps_to_home
    
difference = abs(total_steps - goal)
if total_steps >= goal:
    print("Goal reached! Good job!")
    print(f'{difference} steps over the goal!')
else:
    print(f'{difference} more steps to reach goal.')
